pgml-extension/src/orm/sampling.rs

```diff
@@ -1,11 +1,14 @@
 use pgrx::*;
 use serde::Deserialize;
 
+use super::snapshot::Column;
+
 #[derive(PostgresEnum, Copy, Clone, Eq, PartialEq, Debug, Deserialize)]
 #[allow(non_camel_case_types)]
 pub enum Sampling {
     random,
     last,
+    stratified_random,
```

Comment:
Thinking ahead to other stratification strategies, that use columns other than the target y_column_name, to guarantee you have true out of sample rows. e.g. `user_id` may be used to train with multiple instances from a particular user, but you want to ensure there is no data leakage where the model is just memorizing user_ids rather than the more abstract, so user_id should be excluded as a feature, but used for stratification.

I think that could work as an additional `stratified_column_name`  parameter to train. In that case though, the sampling wouldn't be `stratified_random`. So we'd need to add a different stratified type, or we could just call this `stratified`, and if you don't specify a column, it's random by the y_column_name. If you do specify stratification column(s), then those column(s) get removed from features, and strictly used for stratification.

This can happen in a follow up PR, I'm just commenting so we get a forward looking name on the sampling strategy.

Reply:
Good ideas/points. I changed it to just `stratified`. For this PR, there will be no ability to specify the columns to stratify by, `stratified` only uses y_column_name. I think that should be a non-breaking change to add an optional parameter in the future that would change it from y_column_name to something else.